src/psyclone/tests/psyir/nodes/omp_directives_test.py

```diff
@@ -537,38 +559,64 @@ def test_directive_get_private(monkeypatch):
     # pylint: disable=pointless-statement
     psy.gen
     # Now check that _get_private_clause returns what we expect
-    pvars = directive._get_private_clause()
+    pvars, fpvars = directive._get_private_clauses()
     assert isinstance(pvars, OMPPrivateClause)
+    assert isinstance(fpvars, OMPFirstprivateClause)
     assert len(pvars.children) == 1
+    assert len(fpvars.children) == 0
     assert pvars.children[0].name == 'cell'
-    # Now use monkeypatch to break the Call within the loop
-    call = directive.dir_body[0].dir_body[0].loop_body[0]
-    monkeypatch.setattr(call, "local_vars", lambda: [""])
-    with pytest.raises(InternalError) as err:
-        _ = directive._get_private_clause()
-    assert ("call 'testkern_w3_code' has a local variable but its name is "
-            "not set" in str(err.value))
 
     directive.children[1] = OMPDefaultClause(
             clause_type=OMPDefaultClause.DefaultClauseTypes.NONE)
     with pytest.raises(GenerationError) as excinfo:
-        _ = directive._get_private_clause()
+        _ = directive._get_private_clauses()
     assert ("OMPParallelClause cannot correctly generate the private clause "
             "when its default data sharing attribute in its default clause is "
             "not shared." in str(excinfo.value))
 
 
+def test_directive_get_private(fortran_reader):
+    ''' Tests for the _get_private_clauses() method of OMPParallelDirective.
+    '''
+
+    # Example with private and firstprivate variables
+    psyir = fortran_reader.psyir_from_source('''
+        subroutine my_subroutine()
+            integer :: i, scalar1, scalar2
+            real, dimension(10) :: array
+            do i = 1, 10
```

Comment:
Strictly, you need to give `scalar1` a value before the loop.

Reply:
Done